src/main/java/io/cdap/plugin/servicenow/sink/service/ServiceNowSinkAPIRequestImpl.java

```diff
@@ -0,0 +1,262 @@
+    return restRequest;
+  }
+
+  /**
+   * Inserts/Updates the list of records into ServiceNow table
+   *
+   * @param restRequestsMap The map of restRequests
+   * @param accessToken    The access token
+   */
+  public void createPostRequest(Map<String, RestRequest> restRequestsMap, String accessToken)
+    throws OAuthProblemException, InterruptedException {
+    ServiceNowBatchRequest payloadRequest = getPayloadRequest(restRequestsMap);
+    ServiceNowTableAPIRequestBuilder requestBuilder = new ServiceNowTableAPIRequestBuilder(
+      config.getRestApiEndpoint());
+    RestAPIResponse apiResponse;
+
+    try {
+      requestBuilder.setAuthHeader(accessToken);
+      requestBuilder.setAcceptHeader(MediaType.APPLICATION_JSON);
+      requestBuilder.setContentTypeHeader(MediaType.APPLICATION_JSON);
+      StringEntity stringEntity = new StringEntity(gson.toJson(payloadRequest), ContentType.APPLICATION_JSON);
+      requestBuilder.setEntity(stringEntity);
+      apiResponse = restApi.executePost(requestBuilder.build());
+
+      if (!apiResponse.isSuccess()) {
+        LOG.error("Error - {}", getErrorMessage(apiResponse.getResponseBody()));
+        throw new RetryableException();
+      } else {
+        JsonObject responseJSON = jsonParser.parse(apiResponse.getResponseBody()).getAsJsonObject();
+        JsonArray servicedRequestsArray = responseJSON.get(ServiceNowConstants.SERVICED_REQUESTS).getAsJsonArray();
+        JsonElement failedRequestId = null;
+        for (int i = 0; i < servicedRequestsArray.size(); i++) {
+          int statusCode = servicedRequestsArray.get(i).getAsJsonObject().get(ServiceNowConstants.STATUS_CODE)
+            .getAsInt();
+          if (statusCode == HttpStatus.SC_FORBIDDEN || statusCode == HttpStatus.SC_BAD_REQUEST ||
+            statusCode == HttpStatus.SC_INTERNAL_SERVER_ERROR) {
+            String encodedResponseBody = servicedRequestsArray.get(i).getAsJsonObject().get(ServiceNowConstants.BODY)
+              .getAsString();
+            String decodedResponseBody = new String(Base64.getDecoder().decode(encodedResponseBody));
+            String errorDetail = jsonParser.parse(decodedResponseBody).getAsJsonObject().get(ServiceNowConstants.ERROR)
+              .getAsJsonObject().get(ServiceNowConstants.ERROR_DETAIL).getAsString();
+            
+            if (errorDetail.equals(ServiceNowConstants.ACL_EXCEPTION)) {
+              throw new IllegalStateException(String.format("Permission denied for '%s' operation.",
+                                                            config.getOperation()));
+            } else if (errorDetail.contains(ServiceNowConstants.INSERTION_ERROR)) {
+              LOG.warn("Error Response : {} ", decodedResponseBody);
+            } else if (errorDetail.contains((ServiceNowConstants.MAXIMUM_EXECUTION_TIME_EXCEEDED))) {
+              failedRequestId = servicedRequestsArray.get(i).getAsJsonObject().get(ServiceNowConstants.ID);
+            }
```

Comment:
Don't you need another else clause to handle any other exceptions that might happen and throw an exception in that case?

Reply:
Added an `else` condition throwing exception to handle all other errors returned by ServiceNow Rest API.  383670aa689fe2c05e336d3da0fc2b727f94a760